pkg/ssh/utils.go

```diff
@@ -33,9 +34,9 @@ func Validate(user *url.Userinfo, path string, port int, identity string) (*conf
 
 	if uri.Port() == "" {
```

Comment:
What happens if both `uri.Port()` and `port` is set?

---

Looking at various Podman callers, so many (but not all of them) call `url.Parse` themselves already. Why do that twice, and then need to worry about these port/port conflicts  that might not be actually possible in practice?

Maybe there should be a single `ssh.Destination` type, containing all relevant information in a native support optimized for further use (whether that’s an `url.URL`, individual values, or some combination), and helpers vaguely like `DestinationFromConfigDestination(*config.Destination)`, `DestinationFromURI`, `DestinationFromCLIString`, `DestinationFromPodmanConnectionAddComponents` — each with appropriate heuristics, without several round-trips back and forth.